flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/ApiExpressionUtils.java

```diff
@@ -85,17 +83,7 @@ public static Expression toMonthInterval(Expression e, int multiplier) {
 		// check for constant
 		return ExpressionUtils.extractValue(e, BasicTypeInfo.INT_TYPE_INFO)
 			.map((v) -> (Expression) valueLiteral(v * multiplier, TimeIntervalTypeInfo.INTERVAL_MONTHS))
-			.orElse(
```

Comment:
This changes the behavior of this method right? Previously e.g. `(1 + 3).toMonthInterval` or even `(1 + f0).toMonthInterval` was possible right? (Even though e.g. the window aggregation did not accept it)

Reply:
Yes, that's right but this behavior was not even tested somewhere and for simplification of the design I think it is ok to limit the values to constants. What do you think?

Reply:
I agree, just wanted to double check.